packages/addons/service/docker/source/default.py

```diff
@@ -187,56 +188,56 @@ def print_notification(json_data):
+                                '|',
+                                __addon__.getLocalizedString(30012),
+                                __addon__.getLocalizedString(event_string)])
         except KeyError as e:
-            message = unicode(' '.join([__addon__.getLocalizedString(30011),
-                                        json_data['Type'],
-                                        '|',
-                                        __addon__.getLocalizedString(30012),
-                                        __addon__.getLocalizedString(event_string)]))
+            message = ' '.join([__addon__.getLocalizedString(30011),
+                                json_data['Type'],
+                                '|',
+                                __addon__.getLocalizedString(30012),
+                                __addon__.getLocalizedString(event_string)])
 
     elif __addon__.getSetting('notifications') is '2': # none
         pass
 
     elif __addon__.getSetting('notifications') is '3': # custom
         if __addon__.getSetting(json_data['Action']) == 'true':
             try:
-                message = unicode(' '.join([__addon__.getLocalizedString(30010),
-                                            json_data['Actor']['Attributes']['name'],
-                                            '|',
-                                            __addon__.getLocalizedString(30012),
-                                            __addon__.getLocalizedString(event_string)]))
+                message = ' '.join([__addon__.getLocalizedString(30010),
+                                    json_data['Actor']['Attributes']['name'],
+                                    '|',
+                                    __addon__.getLocalizedString(30012),
+                                    __addon__.getLocalizedString(event_string)])
             except KeyError as e:
-                message = unicode(' '.join([__addon__.getLocalizedString(30011),
-                                            json_data['Type'],
-                                            '|',
-                                            __addon__.getLocalizedString(30012),
-                                            __addon__.getLocalizedString(event_string)]))
+                message = ' '.join([__addon__.getLocalizedString(30011),
+                                    json_data['Type'],
+                                    '|',
+                                    __addon__.getLocalizedString(30012),
+                                    __addon__.getLocalizedString(event_string)])
 
     dialog = xbmcgui.Dialog()
     try:
         if message is not '':
             length = int(__addon__.getSetting('notification_length')) * 1000
             dialog.notification('Docker', message, '/storage/.kodi/addons/service.system.docker/resources/icon.png', length)
-            xbmc.log('## service.system.docker ## ' + unicode(message))
+            xbmc.log('## service.system.docker ## %s' % message)
```

Comment:
Maybe you can use .format for this for python3?
```python
xbmc.log('## service.system.docker ## {}'.format(message))
```

Reply:
Could do, I guess I'm a bit old skool. :smile:

Reply:
We could also use:
```
xbmc.log(f'## service.system.docker ## {message}')
```
There are plenty of ways to skin the string concatenation cat in Python 3.6+ - it all comes down to personal preference, and I suppose I tend to go with the most basic solution. :smile:

Reply:
Basic is always good :smile: